apps/backend/src/app/api/latest/auth/sessions/crud.tsx

```
@@ -0,0 +1,107 @@
+      SELECT data->>'sessionId' as "sessionId", 
+             MAX("eventStartedAt") as "lastActiveAt", 
+             data->>'endUserIpInfoGuess' as "endUserIpInfoGuess", 
+             data->>'isEndUserIpInfoGuessTrusted' as "isEndUserIpInfoGuessTrusted"
+      FROM "Event" 
+      WHERE data->>'sessionId' = ANY(${Prisma.sql`ARRAY[${Prisma.join(refreshTokenObjs.map(s => s.id))}]`})
+      AND "systemEventTypeIds" @> '{"$session-activity"}'
+      GROUP BY data->>'sessionId', data->>'endUserIpInfoGuess', data->>'isEndUserIpInfoGuessTrusted'
+    `;
+
+
+    const sessionsWithLastActiveAt = refreshTokenObjs.map(s => {
+      const event = events.find(e => e.sessionId === s.id);
+      return {
+        ...s,
+        last_active_at: event?.lastActiveAt.getTime(),
+        last_active_at_end_user_ip_info: (event?.endUserIpInfoGuess ? (JSON.parse(event.endUserIpInfoGuess) as GeoInfo)  : undefined),
+      };
+    });
+
+
+    return {
+      items: sessionsWithLastActiveAt.map(s => ({
+        id: s.id,
+        user_id: s.projectUserId,
+        created_at: s.createdAt.getTime(),
+        last_used_at: s.last_active_at,
+        is_impersonation: s.isImpersonation,
+        last_used_at_end_user_ip_info: s.last_active_at_end_user_ip_info,
+        is_current_session: s.id === auth.refreshTokenId,
+      })),
+      is_paginated: false,
+    };
+  },
+  onDelete: async ({ auth, params }: { auth: SmartRequestAuth, params: { id: string }, query: { user_id?: string } }) => {
+
+    const session = await prismaClient.projectUserRefreshToken.findFirst({
+      where: {
+        tenancyId: auth.tenancy.id,
+        id: params.id,
+      },
+    });
+
+    if (!session) {
+      throw new StatusError(StatusError.NotFound, 'Session not found.');
+    }
+
+    if (auth.type === 'client' && auth.user?.id !== session.projectUserId) {
```

maybe this should throw "session not found" error instead as the client shoulnd't get the information if an session from another user exist

---

changed  it